### DISCUSSION TITLE
PVE cluster witness

### DISCUSSION THREAD

**Discussion by AleRousse** [lae/ansible-role-proxmox#161] [Ideas]
So in my knowledge  there is currently no way to tag one host part of the cluster as a witness only. This mean that the tagged host will not be running full blown PVE but only corosync-qdevice to be an arbitrator of the cluster This would add reliability to a 2-nodes cluster. 

[https://www.mankier.com/8/corosync-qdevice](https://www.mankier.com/8/corosync-qdevice)

I would be able to make a pull request with the feature if you think it would be a good idea to support it here. 
What do you think ?

**Comment by lae** [comment]
Hm, I'm familiar with that pattern, but it's possible it might be a really big change.

Before working on it, I'd like to plan this out more so that we're on the same page. Do you have any idea how you think you might implement this?

**Comment by AleRousse** [comment]
Sure, 
I was thinking simply a bout a "pve_cluster_witness: yes"  varriable tagged on the host serving as a witness. 
This variable would be a condition to most of task making sure PVE would not be install on it.
Finnally I would add a specific task to host mark as witness making them join only the corosync cluster as a witness only (can't be voted to be master, only used it split-brain)

**Comment by AleRousse** [comment]
Information about Corosync QDevice(Quorum only) can be found to the following URL: 
https://www.systutorials.com/docs/linux/man/8-corosync-qdevice/
https://github.com/corosync/corosync-qdevice/wiki/Design

Let me know what you think @lae

**Comment by lae** [comment]
I'm not sure adding a condition to most tasks would be a good idea since it sounds like it'd probably increase maintenance (and readability) burden just for a less used feature.

Can you identify which tasks files would be needed to run on witness-only hosts, as well as which tasks in `main.yml`? Maybe we could just move the PVE installation tasks to a separate file and import it in `main.yml` if the host isn't a witness, but I'm wondering how much of this current role would need to run on the witness host in the first place.
